src/net/udp.rs

```diff
@@ -30,16 +32,16 @@ impl UdpSocket {
         if len > 0 {
             // TODO: Remove unwrap and funnel result error types
             let raw_packet: RawPacket = deserialize(&self.recv_buffer[..len]).unwrap();
-            let packet = self.state.process_received(_addr, &raw_packet);
+            let packet = self.state.process_received(_addr, &raw_packet).unwrap();
             Ok(Some(packet))
         } else {
             Ok(None)
         }
     }
 
-    pub fn send(&mut self, packet: Packet) -> io::Result<usize> {
-        let (addr, payload) = self.state.pre_process_packet(packet);
-        self.socket.send_to(&payload, addr)
+    pub fn send(&mut self, packet: Packet) -> Result<io::Result<usize>, AmethystNetworkError> {
+        let (addr, payload) = self.state.pre_process_packet(packet)?;
+        Ok(self.socket.send_to(&payload, addr))
     }
 
     pub fn set_nonblocking(&mut self, nonblocking: bool) -> io::Result<()> {
```

Comment:
I assume this is still a test function?

Reply:
Hm, I didn't write it, but I don't think it is. Can you elaborate @TimonPost ?

Reply:
Sorry saw it wrong, was reviewing on my phone thought there was an incomplete function.